modules/core/src/main/scala/sangria/execution/FutureResolver.scala

```diff
@@ -1572,8 +1691,7 @@ private[execution] class FutureResolver[Ctx](
               deferredResolverState = deferredResolverState
             )
 
-            if (allFields.exists(_.deprecationReason.isDefined))
-              deprecationTracker.deprecatedFieldUsed(ctx)
+            trackDeprecation(ctx)
```

Comment:
The deprecation logic seems more expensive now.
If we're using `DeprecationTracker.empty`, we could actually avoid this work.
What do you think about making `DeprecationTracker` optional in `Executor` (as we're breaking changes anyway), and avoid doing all this work is it's `None`?

Reply:
good idea!